src/components/ChangeData/ChangePassword.tsx

```diff
@@ -0,0 +1,65 @@
+import { FormEvent, useState } from 'react';
+
+import { trpc } from '../../utils/trpc';
+import { TextField } from '../TextField/TextField';
+import { Button } from '../Button/Button';
+import style from './ChangePassword.module.scss';
+
+type ChangePasswordProps = {
+  setWarningMessage: (value: string) => void;
+  setIsVisible: (value: boolean) => void;
+};
+
+export const ChangePassword: React.FC<ChangePasswordProps> = ({
+  setWarningMessage,
+  setIsVisible,
+}) => {
+  const [updatedPassword, setUpdatedPassword] = useState('');
+  const [updatedPasswordControl, setUpdatedPasswordControl] = useState('');
+
+  const updatePassword = trpc.user.updatePassword.useMutation({
+    onSuccess: () => {
+      setWarningMessage('Heslo bylo změněno');
+    },
+    onError: () => {
+      setWarningMessage('Úprava se nezdařila');
+    },
+  });
+
+  const handleSubmitPassword = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (updatedPassword !== updatedPasswordControl) {
+      setWarningMessage('hesla se neshodují');
+      setIsVisible(true);
+    }
```

Comment:
Make sure you return here. So it won't fire the mutation if passwords are not matching :)